plugins/modules/idrac_reset.py

```diff
@@ -92,41 +175,340 @@
   }
 '''
 
-
+import os
 import json
-from ansible_collections.dellemc.openmanage.plugins.module_utils.dellemc_idrac import iDRACConnection, idrac_auth_params
+import time
+from urllib.error import HTTPError, URLError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
 from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from ansible.module_utils.compat.version import LooseVersion
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import (
+    get_idrac_firmware_version, remove_key, get_dynamic_uri, validate_and_get_first_resource_id_uri, idrac_redfish_job_tracking)
+
+
+MANAGERS_URI = "/redfish/v1/Managers"
+OEM = "Oem"
+MANUFACTURER = "Dell"
```

Comment:
This is not proper way to do dynamic uri fetching, here you are hardcoding url in multiple variables and using same below. 

In dynamic uri fetching, you should have only base uri and then reach to target uri dynamically. It'll helps in future if any url changes. It won't break the code.

Please refer to module like idrac_network_attributes, where uri is being fetched dynamically.

Reply:
It's fetching dynamically, but one suggestion, only keep variables of those which is being used in more than one place like ACTIONS, OEM etc., if any variable used in single place, put it directly in code, no need to create variable for single use.

Reply:
Updated accordingly